### PR TITLE
Add example ts figure and restructure

### PR DESCRIPTION
**Pull request by jeromekelleher**
This changes the structure of the Result section a little, by creating a "Tree sequences" section right at the top and moving the content that was in the "data interchange" section in here, with an example tree sequence figure.

I've also moved the "Mutations" section up to the top, because I think it'll get lost in all the ancestry stuff otherwise, and this makes it easier to emphasis the distinction between mutations and ancestry. I've made a note that we should update the rroadmapping at the top, which is out of date.

@petrelharp - is it OK to use your example tree sequence figure here like this?

What do we think of the structure?

### CODE REVIEW

**Comment by petrelharp** [comment]
> @petrelharp - is it OK to use your example tree sequence figure here like this?

Yes! This is the one that [appeared in the treestats paper](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006581), right? (It's at least modified from that?), and that article says:
```
This is an open access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.
```

File: paper.tex [hunk 1]
```diff
@@ -136,23 +136,270 @@ \section*{Introduction}
...
+derived from the 2016 paper and have been extracted from msprime
+into a separate C/Python library.}
 
+\begin{figure}
+\begin{center}
+\includegraphics[width=6cm]{figures/example_tree_sequence}
+\end{center}
+\caption{\label{fig-ts-example} Example tree sequence.
+An example tree sequence describing genealogies and genotypes
+for four samples at ten sites.
```
Comment (petrelharp) [inline paper.tex:185]:
```suggestion
for four samples at ten sites. Figure from \citet{kelleher2018efficient}.
```

**Comment by jeromekelleher** [comment]
I think it's probably different enough that we don't need to bother, but I've updated to say it's "based on" that figure.

**Comment by jeromekelleher** [comment]
OK, nobody has complained so I'm going to merge this. Text is still WIP obvs

**Pull request merged**